omod/src/main/java/org/openmrs/module/patientqueueing/web/resource/PatientQueueResource.java

```diff
@@ -66,6 +72,39 @@ public PatientQueue getByUniqueId(String uniqueId) {
 		return patientQueue;
 	}
 	
+	@Override
+	public Object update(String uuid, SimpleObject propertiesToUpdate, RequestContext context) throws ResponseException {
+		String status=propertiesToUpdate.get("status");
+		if (status == null) {
+			return super.update(uuid, propertiesToUpdate, context);
+		}
+		PatientQueue patientQueue = getPatientQueueForUpdate(uuid, propertiesToUpdate);
+		ValidateUtil.validate(patientQueue);
+		patientQueue.setDateChanged(new Date());
+
+        if (status.equals("picked")) {
```

Comment:
What if someone sets the status and provides the date completed or picked? Do you still want to overwrite it with new Date()?

Reply:
@dkayiwa  done responding to comments